client/app/components/composites/SearchBar/SearchBar.css

```diff
@@ -45,137 +120,277 @@
    next to each other. */
 .keywordInput + .locationInput {
-  border-left: 1px solid var(--SearchBar_borderColor);
-  padding-left: calc(2 * var(--SearchBar_textPadding));
-  margin-right: var(--SearchBar_inputButtonMargin);
+  margin-left: calc(var(--SearchBar_sidePaddingMobile) / 2);
+  margin-right: calc(var(--SearchBar_mobileHeight) + var(--SearchBar_sidePaddingMobile));
+
+  @media (--SearchBar_medium) {
+    border-left: 1px solid var(--SearchBar_borderColor);
+    padding-left: var(--SearchBar_sidePadding);
+    margin-left: 0;
+    margin-right: var(--SearchBar_height);
+  }
 }
 
 .searchButton {
   /* Position absolutely instead of in the flexbox to avoid animation jank. */
   position: absolute;
   top: 0;
   right: 0;
-  bottom: 0;
   margin: 0;
-  padding: 0 var(--SearchBar_sidePadding);
-  padding-top: 2px; /* Fix position. */
-  line-height: var(--SearchBar_searchButtonLineHeight);
+  padding: 0;
+  line-height: 1;
   border: none;
-  background-color: transparent;
+  background-color: var(--SearchBar_mobileButtonBackgroundColor);
+  width: var(--SearchBar_mobileHeight);
+  height: var(--SearchBar_mobileHeight);
   cursor: pointer;
   z-index: var(--SearchBar_childZIndex);
 
-  & svg {
-    line-height: 1;
-    width: 1em;
-    height: 1em;
-    stroke: var(--SearchBar_iconColor);
-    transition: stroke 0.1s ease-in;
+  &:hover,
+  &:active {
+    background-color: var(--SearchBar_mobileButtonBackgroundColor);
+  }
 
-    @nest .root:hover & {
-      stroke: var(--SearchBar_iconColorActive);
+  @media (--SearchBar_medium) {
+    width: var(--SearchBar_height);
+    height: var(--SearchBar_height);
+    background-color: transparent;
 
-      &:hover {
+    &:focus {
+      outline: none;
+
+      & svg {
+        outline: 1px solid var(--SearchBar_borderColor);
         stroke: var(--SearchBar_iconColorFocus);
       }
     }
+
+    @nest .form:hover & {
+
+      & svg {
+        stroke: var(--SearchBar_iconColorActive);
+      }
+
+      &:hover svg,
+      &:focus svg {
+        stroke: var(--SearchBar_iconColorFocus);
+      }
+    }
+
+    &:hover,
+    &:active {
+      background-color: transparent;
+    }
   }
 
-  @nest .root:hover &:hover svg {
-    stroke: var(--SearchBar_iconColorFocus);
+  /* CSS triangle pointing to the search icon above */
+  &::before {
+    content: '';
+    position: absolute;
+    top: calc(-2 * var(--SearchBar_mobileTriangleSize));
+    right: calc(var(--SearchBar_mobileHeight) / 2 - var(--SearchBar_mobileTriangleSize));
+    border: var(--SearchBar_mobileTriangleSize) solid transparent;
+    border-bottom-color: var(--SearchBar_mobileButtonBackgroundColor);
+
+    @media (--SearchBar_medium) {
+      display: none;
+    }
   }
 
-  &:focus {
-    outline: none;
+  & svg {
+    position: absolute;
+    top: calc(var(--SearchBar_mobileHeight) / 2 - var(--SearchBar_iconSize) / 2);
+    left: calc(var(--SearchBar_mobileHeight) / 2 - var(--SearchBar_iconSize) / 2);
```

Comment:
Another way to position SVG elements is: http://tutorials.jenkov.com/svg/svg-viewport-view-box.html#viewbox